lib/new_relic/agent/pipe_channel_manager.rb

```diff
@@ -247,6 +292,11 @@ def unmarshal(data)
         rescue StandardError => e
           ::NewRelic::Agent.logger.error "Failure unmarshalling message from Resque child process, pid #{Process.pid}, size #{data.size}", e
           ::NewRelic::Agent.logger.error Base64.encode64(data)
+          if nr_debug? && File.exist?('/tmp/marshall_log')
+            content = File.binread('/tmp/marshall_log')
+            ::NewRelic::Agent.logger.error "Sent #{content.size}"
```

Comment:
If this is all we're going to do with our file, we can just open an unnamed inode, if ruby exposes those POSIX APIs that is.  They'll get cleaned up automatically, they'll be synced to disk only lazily as the kernel reclaims VFS cache, we won't have to worry about file name conflicts, etc.

Reply:
I agree this is not really reliable. But I'm seeing datas.